### PR TITLE
Fix namespaces

### PR DESCRIPTION
**Pull request by kbaley**
Connects to #61 

Convert root namespace to `NServiceBus.Persistence.AzureStorage`

I haven't renamed the assembly or the project yet. Would like to get a review of what's in progress.

### CODE REVIEW

**Comment by kbaley** [comment]
There are some classes that still need to be moved but we need to wait for the results of: https://github.com/Particular/PlatformDevelopment/pull/759

The blocking issue: The PersistenceDefinition class is called `AzureStoragePersistence` which matches both the convention used in NH and RavenDB and the project namespace itself. So we'll either have to rename the class or change the namespace. The former makes it harder to discover and the latter has more far-reaching ramifications but feels more correct.

**Comment by kbaley** [comment]
This is unblocked now that we've settled on `NServiceBus.Persistence.AzureStorage` as the root namespace.

**Comment by kbaley** [comment]
Ping @Particular/azure-storage-persistence-maintainers

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# NServiceBus.AzureStoragePersistence
+# NServiceBus.Persistence.AzureStorage
 The official [NServiceBus](https://github.com/Particular/NServiceBus) persistence implementation for [Azure Storage](https://azure.microsoft.com/en-us/services/storage/).
 
 Learn more about NServiceBus.AzureStoragePersistence through our [documentation](http://docs.particular.net/nservicebus/azure-storage-persistence/).
```
Comment (dbelcham) [inline README.md]: project name is wrong here

**Comment by dbelcham** [comment]
this gets a :shipit: from me.

Needs 1 more set of @Particular/azure-storage-persistence-maintainers eyes on it before merging

**Comment by SimonCropp** [comment]
Look good to me. Ping me if u need help fixing the doco

**Pull request merged**